### DISCUSSION TITLE
Only imports rewriter

### DISCUSSION THREAD

**Discussion by jogibear9988** [lit/lit#2368] [Q&A]
As the importmaps are not yet here, is there a tool to only rewrite the imports in a web project?
I've done it like this: https://github.com/node-projects/web-component-designer-demo/blob/cf2b6b2770b03040c434b661bc4d53ecea6d818b/gulpfile.cjs#L8

But I think it does not work correctly every time.

**Comment by thescientist13** [comment]
Not quite sure what you're looking for so please feel free to share more details / examples if I am misunderstanding, but there is a library to provide a _shim_ for import map support that is well supported and maintained and works quite well called [**es-module-shims**](https://github.com/guybedford/es-module-shims). it doesn't actually generate the import map for you, but it does provide the functionality for it in the browser.

**Comment by aomarks** [comment]
You can use [Rollup](https://rollupjs.org/) with the [@rollup/plugin-node-resolve](https://github.com/rollup/plugins/tree/master/packages/node-resolve/#readme) plugin and the [`preserveModules`](https://rollupjs.org/guide/en/#outputpreservemodules) option to do a minimal transform that only rewrites bare module specifier imports to paths, keeping your file layout and code identical.

For example, assuming your entrypoint is a file called `index.js`, this command will emit a copy of your JS tree and its dependencies into the `build` folder, with all imports transformed to paths:

```sh
npm i -D rollup @rollup/plugin-node-resolve
npx rollup -i index.js -d build -p @rollup/plugin-node-resolve --preserveModules
```

Note you can specify multiple entrypoints. Rollup also supports a JSON config or a JS API that you may prefer to use over this command-line invocation, see its docs for more.
